node/client/benches/client_bench.rs

```diff
@@ -83,17 +88,15 @@ fn produce_blocks(batches: &mut Vec<Vec<SignedTransaction>>, client: &mut Client
             transactions = batches.remove(0);
         }
         let payload = ChainPayload { transactions, receipts: prev_receipt_blocks };
-        if let BlockProductionResult::Success(_beacon_block, shard_block) =
-            client.try_produce_block(next_block_idx, payload)
-        {
-            prev_receipt_blocks = client
-                .shard_client
-                .get_receipt_block(shard_block.index(), shard_block.shard_id())
-                .map(|b| vec![b])
-                .unwrap_or(vec![]);
-        } else {
-            panic!("Block production should always succeed");
-        }
+        let (beacon_block, shard_block, shard_extra) =
+            client.prepare_block(next_block_idx, payload);
+        let (_beacon_block, shard_block) =
```

Comment:
why `_beacon_block` but `shard_block`? 
also what's the point of returning the blocks you just passed as arguments?